src/pybind/mgr/orchestrator/module.py

```diff
@@ -762,15 +762,25 @@ def _apply_mds(self, fs_name, placement=None, unmanaged=False):
         'orch apply rgw',
         'name=realm_name,type=CephString '
         'name=zone_name,type=CephString '
+        'name=port,type=CephInt,req=false '
+        'name=ssl,type=CephBool,req=false '
         'name=placement,type=CephString,req=false '
         'name=unmanaged,type=CephBool,req=false',
         'Update the number of RGW instances for the given zone')
-    def _apply_rgw(self, zone_name, realm_name, placement=None, unmanaged=False):
+    def _apply_rgw(self, zone_name, realm_name,
+                   port=80,
+                   ssl=False,
+                   placement=None,
+                   unmanaged=False):
+        if not port:
+            port = 443 if ssl else 80
```

Comment:
```suggestion
                   port=None,
                   ssl=False,
                   placement=None,
                   unmanaged=False):
```

can we leave the defaults to the RGWSpec?